src/databricks/labs/ucx/assessment/assessment.py

```diff
@@ -60,20 +68,138 @@ def _verify_ws_client(w: WorkspaceClient):
+        for j in all_jobs:
+            if j.settings.job_clusters is not None:
+                for jc in j.settings.job_clusters:
+                    if jc.new_cluster is None:
+                        continue
+                    yield j, jc.new_cluster
+
+            for t in j.settings.tasks:
+                if t.existing_cluster_id is not None:
+                    interactive_cluster = all_clusters_by_id.get(t.existing_cluster_id, None)
+                    if interactive_cluster is None:
+                        continue
+                    yield j, interactive_cluster
+
+                elif t.new_cluster is not None:
+                    yield j, t.new_cluster
+
+    def _crawl(self) -> list[JobInfo]:
+        all_jobs = list(self._ws.jobs.list(expand_tasks=True))
+        all_clusters = {c.cluster_id: c for c in self._ws.clusters.list()}
+        return self._assess_jobs(all_jobs, all_clusters)
+
+    def _assess_jobs(self, all_jobs: list[BaseJob], all_clusters_by_id) -> list[JobInfo]:
+        incompatible_spark_config_keys = {
+            'spark.databricks.passthrough.enabled',
+            'spark.hadoop.javax.jdo.option.ConnectionURL',
+            'spark.databricks.hive.metastore.glueCatalog.enabled'
+        }
+        job_assessment = {}
+        job_details = {}
+        for job in all_jobs:
+            job_assessment[job.job_id] = set()
+            job_details[job.job_id] = JobInfo(str(job.job_id), job.settings.name, job.creator_user_name, 1, "")
+
+        for job, cluster_config in \
+                self._get_cluster_configs_from_all_jobs(all_jobs, all_clusters_by_id):
+            support_status = spark_version_compatibility(cluster_config.spark_version)
+            if support_status != 'supported':
+                job_assessment[job.job_id].add(f'not supported DBR: {cluster_config.spark_version}')
+
+            if cluster_config.spark_conf is not None:
+                for k in incompatible_spark_config_keys:
+                    if k in cluster_config.spark_conf:
+                        job_assessment[job.job_id].add(f'unsupported config: {k}')
+
+                for value in cluster_config.spark_conf.values():
+                    if 'dbfs:/mnt' in value or '/dbfs/mnt' in value:
+                        job_assessment[job.job_id].add(f'using DBFS mount in configuration: {value}')
+        for job_key in job_details.keys():
+            job_details[job_key].failures = json.dumps(list(job_assessment[job_key]))
+            if len(job_assessment[job_key]) > 0:
+                job_details[job_key].success = 0
+        return list(job_details.values())
+
+    def snapshot(self) -> list[ClusterInfo]:
+        return self._snapshot(self._try_fetch, self._crawl)
+
+    def _try_fetch(self) -> list[ClusterInfo]:
+        for row in self._fetch(
+                f'SELECT * FROM {self._schema}.{self._table}'
+        ):
+            yield JobInfo(*row)
+
+
+if __name__ == "__main__":
```

Comment:
Remove this commented code, we don't allow commented code, as it's confusing other people

Reply:
Done